core/pom.xml

```diff
@@ -122,6 +122,10 @@
       <groupId>org.pentaho</groupId>
       <artifactId>pentaho-metadata</artifactId>
     </dependency>
+    <dependency>
+      <groupId>pentaho</groupId>
+      <artifactId>pentaho-platform-api</artifactId>
```

Comment:
This is because of the IWebsocketEndpoint. We need to put the implementation for that class inside the core project because it will also be used by the cda karaf assembly. At the same time we need to reference the interface from within pentaho-platform, since it's there where the actual websocket endpoint is implemented.

Reply:
Ok, now I remember it... IWebsocketEndpoint is our common interface. But I'm ok with mirroring the implementation and not having to depend on the platform.

Reply:
Reference removed, by moving the interface into cda-core, and refactoring where it was being used in pentaho platform to reference the Class<?> instead of Class<? implements IWebsocketEndpoint>